07.ls_object/lib/ls_object.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+require 'pathname'
+require_relative 'ls_file_stat'
+require_relative 'screen'
+
+class LsObject
+  def initialize(paths, options)
+    @paths = paths.empty? ? ['.'] : paths
+    @options = options
+  end
+
+  def main
+    @options[:header] = !(@paths.length == 1 || @paths.empty?)
+
+    @screen_src = { '' => [] }
+    create_sorted_path_names(@paths).each do |path_name|
+      stat = LsFileStat.new(path_name)
+      if stat.file?
+        @screen_src[''] << stat
+      else
+        @screen_src[path_name.to_s] = create_stats_in_directory(path_name)
+      end
+    end
+
+    Screen.new(@screen_src, @options).show
+  end
+
+  private
+
+  def create_stats_in_directory(base_path_name)
+    file_paths = Dir.glob('*', (@options[:all_visible] ? File::FNM_DOTMATCH : 0), base: base_path_name.to_s)
```

Comment:
```suggestion
    flags = @options[:all_visible] ? File::FNM_DOTMATCH : 0
    file_paths = Dir.glob('*', flags, base: base_path_name.to_s)
```

変数に入れた方が読みやすいと思います。

ちなみにflagsという名前は公式リファレンスから拝借してます
https://docs.ruby-lang.org/ja/latest/class/Dir.html#S_--5B--5D